src/commands/deploy/getDeployFsPath.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { DialogResponses, IActionContext } from 'vscode-azureextensionui';
+import { configurationSettings, extensionPrefix } from '../../constants';
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { ext } from '../../extensionVariables';
+import { isPathEqual, isSubpath } from '../../utils/pathUtils';
+import * as workspaceUtil from '../../utils/workspace';
+import { getWorkspaceSetting, updateGlobalSetting } from '../../vsCodeConfig/settings';
+
+export async function getDeployFsPath(context: IActionContext, target: vscode.Uri | string | SiteTreeItem | undefined, fileExtension?: string): Promise<string> {
+    context.telemetry.properties.deploymentEntryPoint = target ? 'webAppContextMenu' : 'deployButton';
+    if (target instanceof vscode.Uri) {
+        context.telemetry.properties.deploymentEntryPoint = 'fileExplorerContextMenu';
+        return await appendDeploySubpathSetting(target.fsPath);
+    } else if (typeof target === 'string') {
+        // not sure where this entry point would be
+        return await appendDeploySubpathSetting(target);
```

Comment:
lol

It's for the api
https://github.com/microsoft/vscode-azurefunctions/wiki/Azure-Functions-API-(Preview)#deploy

App service could probably remove the `string` case